tactics/hipattern.ml

```diff
@@ -113,7 +113,7 @@ let match_with_one_constructor sigma style onlybinary allow_rec t =
 	else
 	  let ctyp = Termops.prod_applist sigma (EConstr.of_constr mip.mind_nf_lc.(0)) args in
 	  let cargs = List.map RelDecl.get_type (prod_assum sigma ctyp) in
-	  if not (is_lax_conjunction style) || has_nodep_prod sigma ctyp then
+	  if not (is_lax_conjunction style) || has_nodep_prod sigma (Reductionops.nf_beta sigma ctyp) then
```

Comment:
I think you should nf_beta before destructing ctyp into cargs. That way the cargs will verify the expected property of not depending on each other.

Reply:
Done.